File: MapEditor/MapEditor/ActorProperties.cs
namespace MapEditor
{
    using System.Collections.Generic;

    public enum Animation
    {
        Left,
        Up, 
        Down
    }

    public class ActorFile
    {
        public Dictionary<string, DynamicActor> DynamicActors { get; set; }

        public Dictionary<string, StaticActor> StaticActors { get; set; }
    }

    public class ActorProperties
    {
        public string Name { get; set; }

        public float Scale { get; set; }

        public float Width { get; set; }

        public float Height { get; set; }

        public List<Polygon> Polygons { get; set; }
    }

    public class DynamicActor : ActorProperties
    {
        public float Speed { get; set; }

        public float Health { get; set; }

        public Dictionary<Animation, int> Animations { get; set; }
    }

    public class StaticActor : ActorProperties
    {
        public bool Breakable { get; set; }
    }

    public class Polygon
    {
        public float X { get; set; }

        public float Y { get; set; }
    }
}

File: FirstWing/src/com/gamegear/firstwing/actors/json/ActorProperties.java
package com.gamegear.firstwing.actors.json;

import java.util.List;

public class ActorProperties {
	public String Name;
	
	public float Scale;
	
	public float Width;
	
	public float Height;
	
	public List<Polygon> Polygons;
	
	public ActorProperties(){
	}
	
	public ActorProperties(String name, float scale, float width, float height, List<Polygon> polygons){
		this.Name = name;
		this.Scale = scale;
		this.Width = width;
		this.Height = height;
		this.Polygons = polygons;
	}
}
